Source/Controllers/Preferences/Panes/SPNetworkPreferencePane.m

```diff
@@ -230,73 +306,252 @@ - (void)updateSSHConfigPopUp
+
+        if (count != 0) {
+            // select the currently configured value
+            NSString *currentConfig = [prefs stringForKey:SPSSHConfigFile];
+
+            // if currentConfig is @0, they didn't choose anything in the file chooser
+            // so set to default and update prefs.
+            if ([currentConfig isEqualToString:defaultConfig] || currentConfig.isNumeric == YES) {
+                currentConfig = @"Sequel Ace default";
+                [prefs setObject:[[NSBundle mainBundle] pathForResource:SPSSHConfigFile ofType:@""] forKey:SPSSHConfigFile];
+            }
 
-	NSString *defaultConfig = [[NSBundle mainBundle] pathForResource:SPSSHConfigFile ofType:@""];
-	
-	if (count != 0) {
-		// select the currently configured value
-		NSString *currentConfig = [prefs stringForKey:SPSSHConfigFile];
-		
-		if ([currentConfig isEqualToString:defaultConfig]) {
-			currentConfig = @"Sequel Ace default";
-		}
-		
-		[sshConfigChooser selectItemWithTitle:currentConfig];
-	}
+            [button selectItemWithTitle:currentConfig];
+        }
+    }
+    // knownHostsChooser button tagged with @1 in IB
+    else if(button.tag == 2){
+        NSString *defaultConfig = [prefs stringForKey:SPSSHDefaultKnownHostsFile];
+
+        if (count != 0) {
+            // select the currently configured value
+            NSString *currentConfig = [prefs stringForKey:SPSSHKnownHostsFile];
+
+            // if currentConfig is @0, they didn't choose anything in the file chooser
+            // so set to default
+            if ([currentConfig isEqualToString:defaultConfig] || currentConfig.isNumeric == YES) {
+                currentConfig = @"Sequel Ace default";
+            }
+
+            [button selectItemWithTitle:currentConfig];
+
+            if([currentConfig isEqualToString:NSLocalizedString(@"Use known hosts from ssh config (ADVANCED)", @"Use known hosts from ssh config (ADVANCED)")]){
+                BOOL ret = [self checkSSHConfigFileForUserKnownHostsFile:[prefs stringForKey:SPSSHConfigFile]];
+                SPLog(@"checkSSHConfigFileForUserKnownHostsFile ret: [%hhd]", ret);
+
+                if(ret == NO){
+                    NSString *title = NSLocalizedString(@"ERROR: known hosts (ADVANCED)", @"ERROR: known hosts (ADVANCED)");
+
+                    NSString *message = NSLocalizedString(@"No ssh config file contained UserKnownHostsFile.\n\nPlease check your config files and try again.", @"No ssh config file contained UserKnownHostsFile.\n\nPlease check your config files and try again.");
+
+                    [NSAlert createWarningAlertWithTitle:title message:message callback:nil];
+                }
+                else{
+                    // are the UserKnownHostsFiles RW?
+                    [self checkUserKnownHostsFilesAreWritable];
+                    user_defaults_set_bool_ud(SPSSHConfigContainsUserKnownHostsFile, YES, prefs);
```

Comment:
I see where SPSSHConfigContainsUserKnownHostsFile is being set to YES, are we missing a spot where it should be set to NO if the user goes back to using the default or special selected file options?

Reply:
I thought I had .. let me review.